File: pat1085.py
# attention: test case 4 Time Limit Exceeded


n, p = [int(x) for x in input().split()]
seq = [int(x) for x in input().split()]
seq.sort()
# the maximum number of integers
# that can be chosen to form a perfect subsequence
max_num = 0
temp = 0

for i in range(n):
    for j in range(i + max_num, n):
        if seq[j] <= seq[i] * p:
            temp = j - i + 1
            max_num = max(max_num, temp)
        else:
            break

print(max_num)

File: pat1085.cpp
#include<iostream>
#include<algorithm>
using namespace std;

int main() {

	int n;
	long long p;
	cin >> n >> p;
	int seq[100010];
	for (int i = 0; i < n; i++) {
		cin >> seq[i];
	}

	sort(seq, seq + n);

	int max_num = 0, temp = 0;
	for (int i = 0; i < n; i++) {
		for (int j = i + max_num; j < n; j++) {
			if (seq[j] <= seq[i] * p) {
				temp = j - i + 1;
				max_num = max(temp, max_num);
			} else {
				break;
			}
		}

	}

	cout << max_num << endl;


	return 0;
}